pkg/apis/configuration/validation/validation.go

```diff
@@ -707,6 +738,130 @@ func validateVariable(nVar string, validVars map[string]bool, fieldPath *field.P
 		msg := fmt.Sprintf("'%v' contains an invalid NGINX variable. Accepted variables are: %v", nVar, mapToPrettyString(validVars))
 		allErrs = append(allErrs, field.Invalid(fieldPath, nVar, msg))
 	}
+	return allErrs
+}
+
+func isValidSpecialVariableHeader(header string) []string {
+	errMsgs := validation.IsHTTPHeaderName(strings.Replace(header, "_", "-", -1))
+	if len(errMsgs) >= 1 || strings.Contains(header, "-") {
+		return []string{"a valid HTTP header must consist of alphanumeric characters or '_'"}
+	}
+	return nil
+}
+
+func validateSpecialVariable(nVar string, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	value := strings.SplitN(nVar, "_", 2)
+
+	if strings.HasPrefix(nVar, "arg") {
+		_, err := url.ParseQuery(fmt.Sprintf("%v=%v", value[0], value[1]))
```

Comment:
why not an approach similar to the one in `isArgumentName`?

Reply:
Because I didn't see `isArgumentName`